src/main/scala/com/databricks/spark/xml/package.scala

```diff
@@ -19,10 +19,64 @@ import scala.collection.Map
 
 import org.apache.hadoop.io.compress.CompressionCodec
 
+import org.apache.spark.annotation.Experimental
 import org.apache.spark.sql._
-import com.databricks.spark.xml.util.XmlFile
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
+import org.apache.spark.sql.types.StructType
+
+import com.databricks.spark.xml.util.{InferSchema, XmlFile}
 
 package object xml {
+
+  private def withExpr(expr: Expression): Column = new Column(expr)
+
+  /**
+   * Infers the schema of XML documents as strings.
+   *
+   * @param ds Dataset of XML strings
+   * @return inferred schema for XML
+   */
+  @Experimental
+  def inferSchema(ds: Dataset[String]): StructType =
+    inferSchema(ds, Map.empty[String, String])
+
+  /**
+   * Infers the schema of XML documents as strings.
+   *
+   * @param ds Dataset of XML strings
+   * @param options additional XML parsing options
+   * @return inferred schema for XML
+   */
+  @Experimental
+  def inferSchema(ds: Dataset[String], options: Map[String, String]): StructType =
+    InferSchema.infer(ds.rdd, XmlOptions(options.toMap))
+
+  /**
+   * Parses a column containing a XML string into a `StructType` with the specified schema.
+   *
+   * @param e a string column containing XML data
+   * @param schema the schema to use when parsing the XML string
+   */
+  @Experimental
+  implicit def from_xml(e: Column, schema: StructType): Column =
```

Comment:
If https://github.com/databricks/spark-xml/pull/429/files#r361887418 is addressed, I am sure it doesn't need to be an implicit function.